app/src/hooks/domain/useSecondarySector.ts

```diff
@@ -0,0 +1,45 @@
+import { useMemo } from 'react';
+import { isDefined } from '@togglecorp/fujs';
+
+import { useRequest } from '#utils/restRequest';
+import { type GoApiResponse } from '#utils/restRequest';
+
+export type SecondarySector = NonNullable<GoApiResponse<'/api/v2/secondarysector'>>[number];
+
+type ListProps = {
+    id?: never;
+}
+
+type PropsForId = {
+    id: number;
+}
+
+function useSecondarySector(props?: ListProps): [Array<SecondarySector> | undefined, boolean]
+function useSecondarySector(props: PropsForId): [SecondarySector | undefined, boolean]
+function useSecondarySector(
+    props?: ListProps | PropsForId,
+): [SecondarySector | undefined | Array<SecondarySector> | undefined, boolean] {
```

Comment:
I am not sure if this is the correct implementation for fetching enums/options cc: @frozenhelium @samshara

Reply:
@tnagorra should we add these to the `DomainContext` ?

Reply:
When using hooks, users usually expect that this will only make a single request. But, that won't be the case.

We should add this to the DomainContext and make sure the response is cached.

Reply:
Fetched and stored secondary sectors in `DomainContext`.